packages/txservice/src/provider.ts

```diff
@@ -450,4 +464,85 @@ export class ChainRpcProvider {
   private incrementNonce() {
     this._nonce++;
   }
+
+  private async monitorBlockades() {
```

Comment:
Def not a fan of having this in the ChainRpcProvider, but it may have to do for now. Eventually, would be best to be have a self-contained class for this.